sql/async_udfs.mdx

```diff
@@ -0,0 +1,117 @@
+---
+title: Asynchronous User-Defined Functions
+description: 'Run Async UDFs'
+---
+
+Arroyo's UDFs can be asynchronous.
+This is useful for UDFs that need to make network requests or perform other asynchronous operations,
+and therefore should not block processing. Since asynchronous UDFs are executed in parallel,
+we can maintain high throughput even with longer-running UDFs.
+
+Note that async UDFs should not block (i.e. `await`) as this will block the main processing thread.
+Instead, they should use asynchronous operations or [spawn_blocking](https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html).
+
+To define an asynchronous UDF, use the `async` keyword in the function definition.
+
+## Options
+
+Async UDFs have a few configuration options that can be defined in a special comment at the top of the UDF,
+similar to dependencies, under a `[udfs]` header.
+
+| Option | Default | Description |
+| --- | --- | --- |
+| `async_results_ordered` | `true` | Whether to return results in the order they were requested (ordered) or as they are available (unordered). |
+| `async_max_concurrency` | `100` | The maximum number of concurrent executions of this UDF. |
+| `async_timeout_seconds` | `10` | The maximum number of seconds to wait for a result before timing out. Arroyo currently drops the record if the UDF times out or panics. |
+
+
+## Context
+
+Async UDFs can optionally define a `Context` struct that will be passed to each execution of the UDF.
```

Comment:
Do we need to specify the magic naming convention for the UDF?

Reply:
The udf can be named anything, it's just the `Context` that must be named `Context`, and the `context` parameter. I've made that more clear below.